core/src/main/java/io/parsingdata/metal/expression/True.java

```diff
@@ -26,11 +28,22 @@
 public class True implements Expression {
 
     @Override
-    public boolean eval(final Environment environment, final Encoding encoding) { return true; }
+    public boolean eval(final ParseGraph graph, final Encoding encoding) { return true; }
 
     @Override
     public String toString() {
         return getClass().getSimpleName();
     }
 
+    @Override
+    public boolean equals(Object obj) {
+        return obj != null
+            && getClass() == obj.getClass();
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(true);
```

Comment:
Yeah, but semantically that makes sense right? I mean if it has to collide with _something_...